### DISCUSSION TITLE
Does the library create relations based on primary keys only or any indexes??

### DISCUSSION THREAD

**Discussion by kipchirchirian** [tattersoftware/codeigniter4-relations#43] [Q&A] [answered]
I have two tables: `employees` and `skills`:

**Employees**:

| id | employee_id | employee_number | first_name | last_name |
| -- | -------------- | --------------------- | ----------- | ----------- |
| 1 | abc123 | E0001 | John | Doe |
| 2 | def456 | E0002 | Jane | Doe |

**Skills**:

| id | employee_id | skill_name | yrs_of_experience |
| -- | -------------- | ----------- | -------------------- |
| 7 | def456 | Painting | 3 |
| 8 | def456 | Chess | 1 |
| 9 | abc123 | Surfing | 2 |

**Note**:
1.  For both tables, `id` is the _Primary Key_ and it is an auto-incrementing value.
2. `employee_id` is a unique field, it is the foreign key in the `skills` table and the referenced key in the `employees` table.

I have been using this library for some time but not with the tables built like this i.e. the PK is not the referenced key. Now,
simply doing(as I have always done):

``` php
$employee = $employeeModel->with(['skills'])->where('employee_id', 'def456')->find();
```
does not work and always returns an empty array. I have tried playing around with the `$primaryKey` property in both the model and the entity but nothing seems to work.

Is there a way to set the key to use for relations??

**Comment by MGatner** [comment, marked as answer]
This library uses the schema generated by `Tatter\Schemas`, which I believe uses any index available. If you head over to that repo you could check the code; there are also instructions on how to identify related tables using a manually-provided schema.

**Reply by kipchirchirian** [reply]
Thanks! I had seen the option of manually providing a schema and decided to do that as a last option. Seems like I'll have to go that route anyway.

**Reply by MGatner** [reply]
It's up to you! Schemas are merged so all you are really supplying is the value you want to override. If you want to pursue troubleshooting I would start with the CLI command to output the current schema and see what it has detected from your current setup (there is also a Debug Toolbar tab you can check).

**Reply by kipchirchirian** [reply]
Thank you for that info. I was not aware of these options at all. I will look into them ASAP. Thank you soo much for your support!
